timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -399,6 +401,123 @@ def split_by_time(self, cutoff_time: pd.Timestamp) -> Tuple[TimeSeriesDataFrame,
+            When set to None, the slice includes the end of each time series.
+
+        Returns
+        -------
+        ts_df : TimeSeriesDataFrame
+            A new time series dataframe containing entries of the original time series between start and end indices.
+
+        Example
+        -------
+        .. code-block:: python
+
+            >>> print(ts_dataframe)
+            item_id  timestamp  target
+                  0 2019-01-01       0
+                  0 2019-01-02       1
+                  0 2019-01-03       2
+                  1 2019-01-02       3
+                  1 2019-01-03       4
+                  1 2019-01-04       5
+                  2 2019-01-03       6
+                  2 2019-01-04       7
+                  2 2019-01-05       8
+
+            >>> df.slice_items_by_index(0, 1)  # select the first entry of each time series
+            item_id  timestamp  target
+                  0 2019-01-01       0
+                  1 2019-01-02       3
+                  2 2019-01-03       6
+
+            >>> df.slice_items_by_index(-2, None)  # select the last 2 entries of each time series
+            item_id  timestamp  target
+                  0 2019-01-02       1
+                  0 2019-01-03       2
+                  1 2019-01-03       4
+                  1 2019-01-04       5
+                  2 2019-01-04       7
+                  2 2019-01-05       8
+
+            >>> df.slice_items_by_index(None, -1)  # select all except the last entry of each time series
+            item_id  timestamp  target
+                  0 2019-01-01       0
+                  0 2019-01-02       1
+                  1 2019-01-02       3
+                  1 2019-01-03       4
+                  2 2019-01-03       6
+                  2 2019-01-04       7
+
+            >>> df.slice_items_by_index(None, None)  # copy the entire dataframe
+            item_id  timestamp  target
+                  0 2019-01-01       0
+                  0 2019-01-02       1
+                  0 2019-01-03       2
+                  1 2019-01-02       3
+                  1 2019-01-03       4
+                  1 2019-01-04       5
+                  2 2019-01-03       6
+                  2 2019-01-04       7
+                  2 2019-01-05       8
+
+        """
+        length_per_item = self.length_per_item()
+        # Create a boolean index that selects the correct slice in each timeseries
+        boolean_indicators = []
+        for length in length_per_item:
+            indicator = np.zeros(length, dtype=bool)
+            indicator[start_index:end_index] = True
+            boolean_indicators.append(indicator)
+        index = np.concatenate(boolean_indicators)
+        result = TimeSeriesDataFrame(self[index].copy(), static_features=self.static_features)
+        result._cached_freq = self._cached_freq
+        return result
+
+    def slice_items_by_time(self, start_time: pd.Timestamp, end_time: pd.Timestamp) -> TimeSeriesDataFrame:
```

Comment:
`items` appearing in the function name leads me to think that I can provide a subset of items that I want to slice. Why not `slice_by_time` alone?